### PR TITLE
#RUN-462 Fix gem to working with docker

### PR DESCRIPTION
**Pull request by WozniakMac**
If you try to use Mirage with docker it will fail because:

1. In `Mirage.start (runner.rb:14)` it put default port. After that in `Mirage::Client` this if elsif `(client.rb:15)` will never use custom url because at first it check if there is port and if it could find it  `localhost` is used.
2. In `Runner.start` it use `Mirage.running?(options)` - `(runner.rb:91)` with `Hash` type param. Because of that it will always try to find if app is running as `localhost` not u custom url.

I tried to fix this error with backward compatibility. Classes interface stay almost this same.

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-3.0.14
\ No newline at end of file
+3.0.15
```
Comment (dajvido) [inline VERSION]: When will you configure your editor to keep file with empty line?

File: spec/mirage/client/runner_spec.rb [hunk 2]
```diff
@@ -133,6 +163,5 @@
       Mirage::Runner.should_receive(:new).and_return(runner)
       expect { runner.invoke(:stop, [], options) }.not_to raise_error
     end
-
   end
 end
```
Comment (dajvido) [inline spec/mirage/client/runner_spec.rb]: Worth adding that here too.

File: spec/mirage/client/client_spec.rb [hunk 3]
```diff
@@ -17,7 +17,7 @@
       mirage_url = "http://url.for.mirage"
       Client.new(mirage_url).url.should == mirage_url
 
-      Client.new(:url => mirage_url).url.should == mirage_url
+      Client.new(:url => mirage_url, :port => 9001).url.should == mirage_url
```
Comment (dajvido) [inline spec/mirage/client/client_spec.rb]: I'm confused with this port here. This test doesn't test port at all, so it seems to be unnecessary and left to set it to default value. You test configuration with port in the tests below.

File: VERSION (hunk 1 above)
Comment (WozniakMac) [reply VERSION]: I did it... :D

File: lib/mirage/client/runner.rb [hunk 4]
```diff
@@ -42,12 +42,19 @@ def stop options={:port => []}
...
-      url = options_or_url.kind_of?(Hash) ? "http://localhost:#{options_or_url[:port]}" : options_or_url
+      if options_or_url.kind_of?(Hash)
+        if options_or_url[:url]
+          url = options_or_url[:url]
+        else
+          url = "http://localhost:#{options_or_url[:port]}"
+        end
+      else
+        url = options_or_url
+      end
```
Comment (dajvido) [inline lib/mirage/client/runner.rb]: It's getting really complex and hard to read. Could we move it to separated private method to determine the `url` value and use guard clauses there?

```
...

def running?(options_or_url = {:port => 7001})
  HTTParty.get(mirage_url(options_or_url))
  true
rescue Errno::ECONNREFUSED
  false
end

private

def mirage_url(options_or_url)
  return options_or_url unless options_or_url.kind_of?(Hash)
  return options_or_url[:url] if options_or_url[:url]
  "http://localhost:#{options_or_url[:port]}"
end
```

**Review by dajvido — CHANGES_REQUESTED** [review]

**Review by dajvido — CHANGES_REQUESTED** [review]
For some reason I can't set it as "Request Changes" but only as a comment...

**Review by dajvido — APPROVED** [review]

**Pull request merged**